sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -301,26 +296,47 @@ def _preprocess_X(self, X, *, reset):
         )
         self._preprocessor.set_output(transform="default")
         X = self._preprocessor.fit_transform(X)
+        known_categories = self._check_categories()
```

Comment:
It is not obvious that we are going to check the categories from the underling ordinal encoder of the preprocessor with the name `_check_categories`. I don't think that we need to change the name but it could be worth just a small comment.